packages/react-reconciler/src/__tests__/ReactIsomorphicAct-test.js

```diff
@@ -47,4 +47,35 @@ describe('isomorphic act()', () => {
     });
     expect(root).toMatchRenderedOutput('B');
   });
+
+  // @gate __DEV__
+  test('return value – sync callback', async () => {
+    expect(await act(() => 'hi')).toEqual('hi');
+  });
+
+  // @gate __DEV__
+  test('return value – sync callback, nested', async () => {
+    const returnValue = await act(() => {
+      return act(() => 'hi');
+    });
+    expect(returnValue).toEqual('hi');
+  });
+
+  // @gate __DEV__
+  test('return value – async callback', async () => {
+    const returnValue = await act(async () => {
+      return await Promise.resolve('hi');
```

Comment:
What if you just return the promise without awaiting it? I would expect act to await it and return the resolved value. The act  test above implicitly confirms that behavior, but it would be have a test that just returns a promise.

Reply:
I could though that's really a test of the JavaScript spec as opposed to our implementation. An async function always returns a promise. I only added the await and the `Promise.resolve` for illustration purposes.